table/server/master/src/main/java/alluxio/master/table/DbConfig.java

```diff
@@ -55,153 +97,537 @@ public DbConfig(@JsonProperty("bypass") @Nullable BypassEntry bypassEntry) {
+     * @return table name, null if the entry is a regex entry
+     */
+    @Nullable
+    public String getTable() {
+      return getName();
+    }
+
     @Override
     public boolean equals(Object other) {
       if (this == other) {
         return true;
-      } else if (other == null) {
+      }
+      if (other == null) {
         return false;
-      } else if (getClass() != other.getClass()) {
+      }
+      if (getClass() != other.getClass()) {
         return false;
       }
-      BypassTableEntry entry = (BypassTableEntry) other;
-      return Objects.equals(mTableName, entry.mTableName);
+      return super.equals(other);
     }
 
     @Override
     public int hashCode() {
-      return Objects.hashCode(mTableName);
+      return super.hashCode();
     }
   }
 
   /**
-   * Deserializer of BypassTableEntry
-   *
-   * Enables flexible syntax: either a single table name can be specified, and all belonging
-   * partitions will be bypassed;
-   * or an object of form
-   * {"table": "tableName", "partitions": ["part1", "part2"]}
-   * can be used, and individual partitions can be specified.
+   * Accepts a simple table name, an regular expression, or
+   * an object of form: {"table": "tableName", "partitions": ["part1", "part2"]}
    */
-  public static class BypassTableEntryDeserializer extends JsonDeserializer<BypassTableEntry> {
+  static class TableEntryDeserializer extends JsonDeserializer<TableEntry> {
     @Override
-    public BypassTableEntry deserialize(JsonParser jp, DeserializationContext cxt)
+    public TableEntry deserialize(JsonParser jp, DeserializationContext cxt)
         throws IOException, JsonProcessingException {
       ObjectMapper mapper = (ObjectMapper) jp.getCodec();
       JsonNode node = mapper.readTree(jp);
-      String tableName;
-      Set<String> partitions;
-      if (!node.isTextual() && !node.isObject()) {
-        throw new JsonParseException(mapper.treeAsTokens(node), "invalid syntax");
-      } else if (node.isTextual()) {
-        // single table name, all partitions are bypassed
-        tableName = node.asText();
-        partitions = Collections.emptySet();
-      } else {
-        // a {"table": "table", "partitions": ["part1", "part2"]} object
-        if (!node.hasNonNull("table")) {
-          throw new JsonParseException(mapper.treeAsTokens(node), "missing table name");
-        }
-        tableName = node.get("table").asText();
+      // try deserialize as a `NamePatternEntry` object first
+      try {
+        NamePatternEntryDeserializer deserializer = new NamePatternEntryDeserializer();
+        NamePatternEntry namePatternEntry =
+            deserializer.deserialize(mapper.treeAsTokens(node), cxt);
+        return new TableEntry(namePatternEntry);
+      } catch (JsonProcessingException e) {
+        // ignore, and try deserialize as a `TableEntry` object
+      }
+      if (node == null) {
+        return null;
+      }
```

Comment:
`NamePatternEntryDeserializer` should have handled the `null` case itself, so we check it after the deserialization attempt.